cli/cmd/migrate.go

```diff
@@ -64,39 +79,51 @@ func migrate(cmd *cobra.Command, args []string) error {
 			fmt.Println(err)
 		}
 	}()
-	managedDestinationsClients, err := manageddestination.NewClients(ctx, destinations, destinationOpts...)
+	destinationPluginClients, err := managedplugin.NewClients(ctx, destinationPluginConfigs, opts...)
 	if err != nil {
 		return err
 	}
 	defer func() {
-		if err := managedDestinationsClients.Terminate(); err != nil {
+		if err := destinationPluginClients.Terminate(); err != nil {
 			fmt.Println(err)
 		}
 	}()
 
-	for _, cl := range managedSourceClients {
-		maxVersion, err := cl.MaxVersion(ctx)
+	for _, source := range sources {
+		cl := managedSourceClients.ClientByName(source.Name)
+		versions, err := cl.Versions(ctx)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to get source versions: %w", err)
+		}
+		maxVersion := findMaxVersion(versions)
+		if maxVersion >= 3 {
```

Comment:
Shouldn't this be `if maxVersion <= 3`? Or am I missing something 🤔

Reply:
@erezrokah Yeah I think this PR is not intended to handle SDK v3 yet; is that right @yevgenypats ? We may just want to update the title and description of this PR to make it clear this is mostly a refactor. Or are there any user-facing changes?

Reply:
Yeah I updated the title to `refactor` and removed `v4`. The only user facing change is that this CLI doesn't support an old version for sources which I dropped (`source_v0`). Once we move to single plugin protocol we can also have a schedule of how long we support a given protocol so we have a phase where we introduce a new protocol, add deprecation notice and then remove an old protocol support from the CLI. In any case users still can use an old version of the CLI.